tests/data/Ref_nexus_mpes.log

```diff
@@ -317,9 +318,7 @@ DEBUG - classes:
 NXmpes.nxdl.xml:/ENTRY/data
 NXentry.nxdl.xml:/DATA
 NXdata.nxdl.xml:
-DEBUG - NXmpes.nxdl.xml:/ENTRY/data@energy_indices - [NX_CHAR]
-DEBUG - <<RECOMMENDED>>
-DEBUG - documentation (NXmpes.nxdl.xml:/ENTRY/data/energy_indices):
+DEBUG - @energy_indices - IS NOT IN SCHEMA
```

Comment:
Is this because `AXISNAME(NX_NUMBER)` and `DATA(NX_NUMBER)` in `NXdata`? If there was only one of them, would it get assigned properly? I actually never understood how there can be two unnamed fields of the same type and how that is supposed to be resolved by any reader/validator.

Reply:
No, this is another problem. I think generally if you have a variable field name (i.e. uppercase field) it only collects all concepts which are not defined otherwise. This means whenever we write something down in our appdef we essentially remove it from the variadic concept and we would also need to repeat all the additional concepts (like AXISNAME_indices etc) the variadic field has or we'd need a mechanism to tell that this actually belongs to the `AXISNAME` concept and we're just extending it. We basically need a variadic field name inheritance 😅🙈.

The `AXISNAME` and `DATA` ambiguity is also a problem, I agree. Some of these are resolved by the `@signal` and `@axes` attributes and the namefitting (which checks how many characters match to the concept, but the name `zzzz` would get the same score 0 for both `AXISNAME` _and_ `DATA` so it's not a very helpful concept here).

Reply:
I learned that there is a speciality in NXdata that everything in `@signal` and `@axes` is automatically of concept `DATA` or `AXISNAME`. Therefore, the indices should actually be recognised but this is a shortcoming of the general algorithm and not of the changes here.